google/resource_dns_record_set.go

```diff
@@ -65,6 +162,12 @@ func resourceDnsRecordSetCreate(d *schema.ResourceData, meta interface{}) error
 
 	zone := d.Get("managed_zone").(string)
 
+	// Validate the Recordset name to be added
+	err = validate_dns_rrset_name(d.Get("name").(string))
```

Comment:
If you make this a `ValidateFunc` on the `name` attribute instead of doing the check here, then it'll fail during `terraform plan` instead of during `terraform apply`

Reply:
Done. Had to refactor to return as full a list of errors as possible rather than the first one